File: android/src/main/kotlin/com/tradechart/plugin/TradeChartPlugin.kt
package com.tradechart.plugin

import com.tradechart.plugin.bridge.ChartFlutterApiHolder
import com.tradechart.plugin.bridge.ChartHostApiImpl
import com.tradechart.plugin.bridge.generated.ChartFlutterApi
import com.tradechart.plugin.bridge.generated.ChartHostApi
import io.flutter.embedding.engine.plugins.FlutterPlugin

class TradeChartPlugin : FlutterPlugin {
    private var hostApiImpl: ChartHostApiImpl? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val flutterApiHolder = ChartFlutterApiHolder(ChartFlutterApi(binding.binaryMessenger))
        hostApiImpl = ChartHostApiImpl(
            textureRegistry = binding.textureRegistry,
            flutterApiHolder = flutterApiHolder,
        )
        ChartHostApi.setUp(binding.binaryMessenger, hostApiImpl)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        hostApiImpl?.dispose()
        hostApiImpl = null
        ChartHostApi.setUp(binding.binaryMessenger, null)
    }
}

File: ios/Classes/TradeChartPlugin.swift
import Flutter
import Foundation

public class TradeChartPlugin: NSObject, FlutterPlugin {
  private var hostApiImpl: ChartHostApiImpl?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let instance = TradeChartPlugin()
    let flutterApiHolder = ChartFlutterApiHolder(
      flutterApi: ChartFlutterApi(binaryMessenger: registrar.messenger())
    )
    instance.hostApiImpl = ChartHostApiImpl(
      textureRegistry: registrar.textures(),
      flutterApiHolder: flutterApiHolder
    )
    ChartHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: instance.hostApiImpl)
  }
}
